Packs/GCP-IAM/Integrations/GCPIAM/GCPIAM.py

```diff
@@ -0,0 +1,3715 @@
+    command_results = CommandResults(
+        readable_output=f'Role {role_name} updated successfully.'
+    )
+    return command_results
+
+
+def remove_custom_role_permissions(client_request_get_method: Callable, client_request_update_method: Callable,
+                                   args: Dict[str, Any]):
+    """
+    Remove permissions from custom project role.
+    Args:
+        client_request_get_method (Callable): The GCP Client method which retrieved
+                                              the required resource (organization/project) role.
+        client_request_update_method (Callable): The GCP Client method which update
+                                                 the required resource (organization/project) role.
+        args (dict): Command arguments from XSOAR.
+
+    Returns:
+        CommandResults: outputs, readable outputs and raw response for XSOAR.
+
+    """
+    role_name = args.get('role_name')
+    permissions = argToList(args.get('permissions'))
+
+    role_permissions = client_request_get_method(role_name).get('includedPermissions', [])
+
+    for permission in permissions:
+        try:
+            role_permissions.remove(permission)
+        except ValueError:
+            raise Exception(f'The permission {permission} is not part of the custom resource permissions.'
+                            f'If you wish to add a new permission, '
+                            f'consider using gcp-iam-organization-role-permission-add '
+                            f'or gcp-iam-project-role-permission-add command.')
+
+    client_request_update_method(role_name, permissions=role_permissions, fields_to_update="includedPermissions")
+    command_results = CommandResults(
+        readable_output=f'Role {role_name} updated successfully.'
+    )
+    return command_results
+
+
+def list_filtered_role(client_request_method: Callable, command_arguments: dict, limit: int, page: int,
+                       max_limit: int, title_filter: str = None, permission_filter: list = None) -> tuple:
+    """
+    List and filter roles.
+    Args:
+        client_request_method (Callable): The GCP Client method which list the required resource
+                                          (predefined/organization/project) role.
+        command_arguments (dict): Client method arguments.
+        limit (int): The number of results to retrieve.
+        page (int): The page number of the results to retrieve.
+        max_limit (int): GCP API max limit.
+        title_filter (str): Used to filter the retrieved roles by the rule title.
+        permission_filter (list): Used to filter the retrieved roles by their permissions.
+
+    Returns:
+        response , outputs.
+
+    """
+    if permission_filter:
+        command_arguments["include_permissions"] = True
+
+    response = client_request_method(limit=max_limit, **command_arguments)
```

Comment:
what if page=2 and limit=1000? if I understand correctly it's a valid request but currently you can't bring roles 1001 and above. please check if it's possible to pass the page arg